reflection/serverreflection.go

```diff
@@ -63,9 +64,27 @@ type GRPCServer interface {
 var _ GRPCServer = (*grpc.Server)(nil)
 
 // Register registers the server reflection service on the given gRPC server.
+// Both the v1 and v1alpha versions are registered.
 func Register(s GRPCServer) {
-	svr := NewServer(ServerOptions{Services: s})
-	v1alphagrpc.RegisterServerReflectionServer(s, svr)
+	svr := NewServerV1(ServerOptions{Services: s})
+	v1alphagrpc.RegisterServerReflectionServer(s, asV1Alpha(svr))
+	v1grpc.RegisterServerReflectionServer(s, svr)
+}
+
+// RegisterV1 registers only the v1 version of the server reflection service
+// on the given gRPC server. Many clients may only support v1alpha so most
+// users should use Register instead.
+func RegisterV1(s GRPCServer) {
+	svr := NewServerV1(ServerOptions{Services: s})
+	v1grpc.RegisterServerReflectionServer(s, svr)
+}
+
+// RegisterV1Alpha registers only the v1alpha version of the server reflection
+// service. Some clients may prefer v1 or only support v1 exclusively, so most
+// users should use Register instead.
```

Comment:
Do you think we need this function?  I think we could just not have it at all.  If it's necessary, maybe call it experimental or even temporary?

Reply:
I had just added it for completeness and symmetry with V1. But I can't think of a particularly compelling need outside of that. So I removed it.